### PR TITLE
feat(utils): ✨ add cookie value sanitizer

### PR DESCRIPTION
**Pull request by goztrk**
## Description
- Adds `sanitize_cookie_value` utility function.

### CODE REVIEW

File: utils/text/sanitizers.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    References:
+    - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+    - https://stackoverflow.com/a/1969339
+    """
+    # Escape HTML special characters to prevent XSS
+    sanitized_value = html.escape(value)
+
+    # Further restrict to a safe set of characters
+    # This regular expression allows only alphanumeric characters, hyphens, and underscores
+    sanitized_value = re.sub(r'[^a-zA-Z0-9-_]', '', sanitized_value)
```
Comment (jontsai) [inline utils/text/sanitizers.py:31]: Is this overly strict? Do we want to allow `!#$%&'()*+-./:<=>?@[]^_`{|}~` as mentioned in the StackOverflow article? Do we want to allow spaces?

File: utils/text/sanitizers.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    Returns:
+    - str: The sanitized cookie value.
+
+    References:
+    - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+    - https://stackoverflow.com/a/1969339
+    """
+    # Escape HTML special characters to prevent XSS
+    sanitized_value = html.escape(value)
```
Comment (jontsai) [inline utils/text/sanitizers.py:27]: Really, in a cookie? Honest question.

Comment (goztrk) [reply utils/text/sanitizers.py:27]: It's suggested by ChatGPT and It doesn't hurt to do extra checks to ensure no funny business going on.

File: utils/text/sanitizers.py (hunk 1 above)
Comment (goztrk) [reply utils/text/sanitizers.py]: Updated.

File: utils/text/sanitizers.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    # Further restrict to a safe set of characters
+    sanitized_value = re.sub(
+        r'[^a-zA-Z0-9-_ !#$%&\'()*+-./:<=>?@[\]^_`{|}~]', '', sanitized_value
```
Comment (jontsai) [inline utils/text/sanitizers.py:30]:
```suggestion
        r'[^a-zA-Z0-9-_!#$%&\'()*+-./:<=>?@[\]^_`{|}~]', '', sanitized_value
```
Per S.O., space is also prohibited

> In the cookie value it formally bans the (filtered by browsers) control characters and (inconsistently-implemented) non-ASCII characters. It retains cookie_spec's prohibition on space, comma and semicolon

**Review by jontsai — CHANGES_REQUESTED** [review]

**Review by jontsai — APPROVED** [review]

File: utils/text/sanitizers.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    References:
+    - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+    - https://stackoverflow.com/a/1969339
+    """
+    # Escape HTML special characters to prevent XSS
+    sanitized_value = html.escape(value)
+
+    # Further restrict to a safe set of characters
+    sanitized_value = re.sub(
+        r'[^a-zA-Z0-9-_!#$%&\'()*+-./:<=>?@[\]^_`{|}~]', '', sanitized_value
```
Comment (jontsai) [inline utils/text/sanitizers.py:30]: Hmm the S.O. comments further mention:

image

Seems like spaces _could_ be allowed. I won't make the change at this time, but just keep an eye on it.

**Pull request merged**